src/box/relay.cc

```diff
@@ -931,7 +955,8 @@ relay_subscribe_f(va_list ap)
 		relay_schedule_pending_gc(relay, send_vclock);
 
 		if (vclock_sum(&relay->status_msg.vclock) ==
-		    vclock_sum(send_vclock))
+		    vclock_sum(send_vclock) &&
+		    ev_monotonic_now(loop()) - relay->tx_seen_time <= timeout)
```

Comment:
You probably don't need this. You can send the status message just fine. If it is stuck because of tx thread being unresponsive, you won't send a new one anyway because there is a check above:
```C
if (relay->status_msg.msg.route != NULL)
	continue;
```

Reply:
I was confused by that `if (vclock_sum(...` too at first.
My intention is to skip the message only when less than a second passed after previous tx interaction.
And when more than a second has passed, do not skip.
It's just expressed backwards in this if.

Reply:
But didn't you make by this change so now downstream vclock in TX thread is updated less often? Even when TX is functioning fine. Does the test fail if you remove this diff hunk?

Reply:
> But didn't you make by this change so now downstream vclock in TX thread is updated less often?

No. If I revert this change, and there are no new rows, then relay will always skip sending a new vclock message to tx, and tx won't respond with `tx_seen_time` message, so relay will think tx is dead.

So both with and without this change the test passes, but without this change relay erroneously stops sending heartbeats when there is no replication stream.

I made the condition for skipping the update more strict (by adding `&& another condition`), so this change makes relay **not skip** updates more often

Reply:
Thanks for the explanation, I misread the condition!